src/transformState.js

```diff
@@ -5,7 +5,38 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const type = action.type;
+
+    switch (type) {
+      case 'addProperties':
+        const extraData = action.extraData;
+
+        for (const key in extraData) {
+          state[key] = extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        const keysToRemove = action.keysToRemove;
+
+        for (const key of keysToRemove) {
+          if (state.hasOwnProperty(key)) {
```

Comment:
Redundant check, because the **delete** operator will not return an error if the matching key is not found